File: erezervisi_api/eRezervisi/eRezervisi.Common.Dtos/Reservation/ReservationCreateDto.cs
using eRezervisi.Core.Domain.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace eRezervisi.Common.Dtos.Reservation
{
    public class ReservationCreateDto
    {
        public long AccommodationUnitId { get; set; }
        public DateTime From { get; set; }
        public DateTime To { get; set; }
        public string? Note { get; set; }
        public int NumberOfAdults { get; set; }
        public int NumberOfChildren { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        [NotMapped]
        public int TotalPeople => NumberOfAdults + NumberOfChildren;
    }
}

File: erezervisi_mobile/erezervisi_mobile/lib/models/requests/reservation/reservation_create_dto.dart
import 'package:erezervisi_mobile/enums/payment_method.dart';

class ReservationCreateDto {
  late num accommodationUnitId;
  late DateTime from;
  late DateTime to;
  late String? note;
  late num numberOfAdults;
  late num numberOfChildren;
  late PaymentMethod paymenthMethod;

  ReservationCreateDto({
    required this.accommodationUnitId,
    required this.from,
    required this.to,
    this.note,
    required this.numberOfAdults,
    required this.numberOfChildren,
    required this.paymenthMethod,
  });

  Map<String, dynamic> toJson() {
    return {
      'accommodationUnitId': accommodationUnitId,
      'from': from,
      'to': to,
      'note': note,
      'numberOfAdults': numberOfAdults,
      'numberOfChildren': numberOfChildren,
      'paymenthMethod': paymenthMethod,
    };
  }
}
